src/main/java/com/commercetools/sync/shoppinglists/utils/LineItemUpdateActionUtils.java

```diff
@@ -0,0 +1,257 @@
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+public final class LineItemUpdateActionUtils {
+
+    /**
+     * Compares a list of {@link LineItem}s with a list of {@link LineItemDraft}s.
+     * The method takes in functions for building the required update actions (AddLineItem, RemoveLineItem and
+     * 1-1 update actions on line items (e.g. changeLineItemQuantity, setLineItemCustomType, etc..).
+     *
+     * <p>If the list of new {@link LineItemDraft}s is {@code null}, then remove actions are built for every existing
+     * line item.
+     *
+     * @param oldShoppingList shopping list resource, whose line item should be updated.
+     * @param newShoppingList new shopping list draft, which contains the line item to update.
+     * @param syncOptions     responsible for supplying the sync options to the sync utility method. It is used for
+     *                        triggering the error callback within the utility, in case of errors.
+     * @return a list of line item update actions on the resource of shopping lists, if the list of line items are not
+     *         identical. Otherwise, if the line items are identical, an empty list is returned.
+     */
+    @Nonnull
+    public static List<UpdateAction<ShoppingList>> buildLineItemsUpdateActions(
+        @Nonnull final ShoppingList oldShoppingList,
+        @Nonnull final ShoppingListDraft newShoppingList,
+        @Nonnull final ShoppingListSyncOptions syncOptions) {
+
+        boolean hasOldLineItems = oldShoppingList.getLineItems() != null && !oldShoppingList.getLineItems().isEmpty();
+        boolean hasNewLineItems = newShoppingList.getLineItems() != null && !newShoppingList.getLineItems().isEmpty()
+            && newShoppingList.getLineItems().stream().anyMatch(Objects::nonNull);
+
+        if (hasOldLineItems && !hasNewLineItems) {
+
+            return oldShoppingList.getLineItems()
+                                  .stream()
+                                  .map(RemoveLineItem::of)
+                                  .collect(toList());
+
+        } else if (!hasOldLineItems) {
+
+            if (!hasNewLineItems) {
+                return emptyList();
+            }
+
+            return newShoppingList.getLineItems()
+                                  .stream()
+                                  .filter(Objects::nonNull)
+                                  .map(AddLineItemWithSku::of)
+                                  .collect(toList());
+        }
+
+        final List<LineItem> oldLineItems = oldShoppingList.getLineItems();
+        final List<LineItemDraft> newlineItems = newShoppingList.getLineItems()
+                                                                .stream()
+                                                                .filter(Objects::nonNull)
+                                                                .collect(toList());
```

Comment:
Can move to line 50, so that it is not required to check nonNull element in line 53 and 70. wdyt?

Reply:
It's required, I will keep null checks and array creation separate.